src/mapping/circular_to_cartesian.hpp

```diff
@@ -228,74 +221,45 @@ class CircularToCartesian
+     * @return A double with the value of the (i,j) coefficient of the inverse Jacobian matrix.
      */
-    KOKKOS_FUNCTION double inv_jacobian_12(Coord<R, Theta> const& coord) const
+    template <class IndexTag1, class IndexTag2>
+    KOKKOS_FUNCTION double inv_jacobian_component(Coord<R, Theta> const& coord) const
     {
-        const double theta = ddc::get<Theta>(coord);
-        return Kokkos::sin(theta);
-    }
+        static_assert(ddc::in_tags_v<IndexTag1, VectorIndexSet<R, Theta>>);
+        static_assert(ddc::in_tags_v<IndexTag2, VectorIndexSet<X_cov, Y_cov>>);
 
-    /**
-     * @brief Compute the (2,1) coefficient of the inverse Jacobian matrix.
-     *
-     * Be careful because not all mappings are invertible, especially at the centre point.
-     *
-     * @param[in] coord
-     *              The coordinate where we evaluate the inverse Jacobian matrix.
-     *
-     * @return A double with the value of the (2,1) coefficient of the inverse Jacobian matrix.
-     */
-    KOKKOS_FUNCTION double inv_jacobian_21(Coord<R, Theta> const& coord) const
-    {
-        const double r = ddc::get<R>(coord);
         const double theta = ddc::get<Theta>(coord);
-        assert(fabs(r) >= 1e-15);
-        return -1 / r * Kokkos::sin(theta);
-    }
 
-    /**
-     * @brief Compute the (2,2) coefficient of the inverse Jacobian matrix.
-     *
-     * Be careful because not all mappings are invertible, especially at the centre point.
-     *
-     * @param[in] coord
-     *              The coordinate where we evaluate the inverse Jacobian matrix.
-     *
-     * @return A double with the value of the (2,2) coefficient of the inverse Jacobian matrix.
-     */
-    KOKKOS_FUNCTION double inv_jacobian_22(Coord<R, Theta> const& coord) const
-    {
-        const double r = ddc::get<R>(coord);
-        const double theta = ddc::get<Theta>(coord);
-        assert(fabs(r) >= 1e-15);
-        return 1 / r * Kokkos::cos(theta);
+        if constexpr (std::is_same_v<IndexTag1, R> && std::is_same_v<IndexTag2, X_cov>) {
+            //Compute the (1,1) coefficient of the inverse Jacobian matrix.
+            return Kokkos::cos(theta);
+        } else if constexpr (std::is_same_v<IndexTag1, R> && std::is_same_v<IndexTag2, Y_cov>) {
+            //Compute the (1,2) coefficient of the inverse Jacobian matrix.
+            return Kokkos::sin(theta);
+        } else if constexpr (std::is_same_v<IndexTag1, Theta> && std::is_same_v<IndexTag2, X_cov>) {
+            //Compute the (2,1) coefficient of the inverse Jacobian matrix.
+            const double r = ddc::get<R>(coord);
+            assert(fabs(r) >= 1e-15);
+            return -1 / r * Kokkos::sin(theta);
+        } else {
+            //Compute the (2,2) coefficient of the inverse Jacobian matrix.
+            const double r = ddc::get<R>(coord);
+            assert(fabs(r) >= 1e-15);
+            return 1 / r * Kokkos::cos(theta);
+        }
```

Comment:
Code duplication could be reduced. I'm not sure that this reduces the number of lines though so feel free to pick the version that you find more readable
```suggestion
        } else {
            const double r = ddc::get<R>(coord);
            assert(fabs(r) >= 1e-15);
            if constexpr (std::is_same_v<IndexTag2, X_cov>) {
                //Compute the (2,1) coefficient of the inverse Jacobian matrix.
                return -1 / r * Kokkos::sin(theta);
            } else {
                //Compute the (2,2) coefficient of the inverse Jacobian matrix.
                return 1 / r * Kokkos::cos(theta);
            }
        }
```